When bullet is disabled, drake no longer compiles

These are the error messages:
- /home/jwnimmer/jwnimmer-tri/drake-distro/drake/systems/plants/collision/DrakeCollision.cpp:45:69: error: ‘DRAKE_ABORT_UNLESS’ was not declared in this scope
- /home/jwnimmer/jwnimmer-tri/drake-distro/drake/systems/plants/collision/DrakeCollision.cpp:47:1: error: no return statement in function returning non-void [-Werror=return-type]

--- a/drake/systems/plants/collision/DrakeCollision.cpp
+++ b/drake/systems/plants/collision/DrakeCollision.cpp
@@ -21,7 +21,7 @@ unique_ptr<Model> newModel() {
   return unique_ptr<Model>(new BulletModel());
 #else
   DRAKE_ABORT_UNLESS(!"DrakeCollision must be compiled with Bullet.");
-  // This return statement is actually never reached when 
+  // This return statement is actually never reached when
   // BULLET_COLLISION is defined.
   // It is placed here to avoid a compiler error on not having a return
   // statement.

--- a/drake/systems/plants/collision/DrakeCollision.cpp
+++ b/drake/systems/plants/collision/DrakeCollision.cpp
@@ -21,8 +21,7 @@ unique_ptr<Model> newModel() {
   return unique_ptr<Model>(new BulletModel());
 #else
   DRAKE_ABORT_UNLESS(!"DrakeCollision must be compiled with Bullet.");
-  // This return statement is actually never reached when
-  // BULLET_COLLISION is defined.
+  // This return statement is actually never reached.
   // It is placed here to avoid a compiler error on not having a return
   // statement.
   return unique_ptr<Model>();

--- a/drake/systems/plants/collision/DrakeCollision.cpp
+++ b/drake/systems/plants/collision/DrakeCollision.cpp
@@ -1,8 +1,8 @@
+#include "drake/systems/plants/collision/DrakeCollision.h"
 
 #include <iostream>
 
-#include "drake/systems/plants/collision/DrakeCollision.h"
-
+#include "drake/common/drake_assert.h"
 #ifdef BULLET_COLLISION
 #include "drake/systems/plants/collision/bullet_model.h"
 #endif
@@ -43,6 +43,10 @@ unique_ptr<Model> newModel() {
   return newModel(BULLET);
 #else
   DRAKE_ABORT_UNLESS(!"DrakeCollision must be compiled with Bullet.");
+  // This return statement is actually never reached.
+  // It is placed here to avoid a compiler error on not having a return
+  // statement.
+  return unique_ptr<Model>();
 #endif
 }
 };

--- a/drake/systems/plants/collision/DrakeCollision.cpp
+++ b/drake/systems/plants/collision/DrakeCollision.cpp
@@ -16,34 +16,13 @@ const bitmask ALL_MASK(bitmask(0).set());
 const bitmask NONE_MASK(0);
 const bitmask DEFAULT_GROUP(1);
 
-enum DRAKECOLLISION_EXPORT ModelType { NONE, AUTO, BULLET };
-
-unique_ptr<Model> newModel(ModelType model_type) {
-  switch (model_type) {
-    case NONE:
-      return nullptr;
-      break;
-    case BULLET:
-#ifdef BULLET_COLLISION
-      return unique_ptr<Model>(new BulletModel());
-#else
-      cerr << "Recompile with Bullet enabled (-DBULLET_COLLISION) to use "
-              "Bullet collision models." << endl;
-#endif
-      break;
-    default:
-      cerr << model_type << " is not a recognized collision model type."
-           << endl;
-  }
-  return unique_ptr<Model>();
-}
-
 unique_ptr<Model> newModel() {
 #ifdef BULLET_COLLISION
-  return newModel(BULLET);
+  return unique_ptr<Model>(new BulletModel());
 #else
   DRAKE_ABORT_UNLESS(!"DrakeCollision must be compiled with Bullet.");
-  // This return statement is actually never reached.
+  // This return statement is actually never reached when 
+  // BULLET_COLLISION is defined.
   // It is placed here to avoid a compiler error on not having a return
   // statement.
   return unique_ptr<Model>();

